server/src/main/java/org/apache/druid/segment/handoff/CoordinatorBasedSegmentHandoffNotifier.java

```diff
@@ -114,10 +115,10 @@ void checkForSegmentHandoffs()
         catch (Exception e) {
           log.error(
               e,
-              "Exception while checking handoff for dataSource[%s] Segment[%s], Will try again after [%d]secs",
+              "Exception while checking handoff for dataSource[%s] Segment[%s]; will try again after [%s]",
               dataSource,
               descriptor,
-              pollDurationMillis
+              Duration.ofMillis(pollDurationMillis).toString()
```

Comment:
We might as well just change the field `long pollDurationMillis` to be `Duration pollDuration` instead. The constructor already takes a `Duration`. We could also use `ScheduledExecutors.scheduleAtFixedRate` in line 75 for a `Duration` friendly invocation.